### ISSUE TITLE
HTML Placeholder issue

### ISSUE THREAD

**Issue by rst30** [Amberg/DocxTemplater#23]
Hello,
I am currently trying to insert HTML snippet/code into a template. I set up a very simple template like so :
![image](https://github.com/user-attachments/assets/b88e7d62-d105-4e03-ab87-cb4a21c39d69)

Is this correct usage of the html placeholder?

I am using the code below to generate my file

```
using (MemoryStream generatedDocument = new MemoryStream())
{
    string htmlValue = "insert html text here";
    var templatePath = Path.Combine(_env.ContentRootPath, "InareTemplates", "TEST_TEMPLATE.docx");
    var template = DocxTemplate.Open(templatePath);
    template.BindModel("ds", new { CLAUSES = htmlValue });
    template.Save(generatedDocument);
    return File(generatedDocument.ToArray(), "application/vnd.openxmlformats-officedocument.wordprocessingml.document", "RenderedDocument.docx");
}
```

However, upon running it, it ran into an exception at this line
template.Save(generatedDocument);
With the below error :
`System.NullReferenceException: 'Object reference not set to an instance of an object.'`

Am I using the html template/placeholder wrong? The example in the readme does not explain how to use it with the model.

Here is the html code sample I am using

`<h1>The Main Languages of the Web</h1><p>HTML is the standard markup language for creating Web pages. HTML describes the structure of a Web page, and consists of a series of elements. HTML elements tell the browser how to display the content.</p><hr><p>CSS is a language that describes how HTML elements are to be displayed on screen, paper, or in other media. CSS saves a lot of work, because it can control the layout of multiple web pages all at once.</p><hr><p>JavaScript is the programming language of HTML and the Web. JavaScript can change HTML content and attribute values. JavaScript can change CSS. JavaScript can hide and show HTML elements, and more.</p>`

Kind regards

**Comment by rst30** [comment]
Also would like to add, returning the template into the very basic example in the readme made it work without errors, passing the raw html text inside
Template :
![image](https://github.com/user-attachments/assets/6004a4af-887e-4201-b467-66249fe8715c)
Result :
![image](https://github.com/user-attachments/assets/dab2b3d6-3f89-4038-8565-875ae158aba4)

**Comment by rst30** [comment]
I modified my code to somewhat follow your example in the tests folder,
```
using (MemoryStream generatedDocument = new MemoryStream())
{
    string htmlValue = htmlContent; 
    htmlValue = "html code here";

    var templatePath = Path.Combine(_env.ContentRootPath, "InareTemplates", "TEST_TEMPLATE.docx");
    var template = DocxTemplate.Open(templatePath);
    template.BindModel("ds", htmlValue);
    template.Process();
    template.Validate();
    template.Save(generatedDocument);
    return File(generatedDocument.ToArray(), "application/vnd.openxmlformats-officedocument.wordprocessingml.document", "RenderedDocument.docx");
}

```
This is the error that i could glance
```
at DocxTemplater.Formatter.VariableReplacer.ApplyFormatter(PatternMatch patternMatch, ValueWithMetadata valueWithMetadata, Text target)
   at DocxTemplater.Formatter.VariableReplacer.ReplaceVariables(OpenXmlElement cloned)
   at DocxTemplater.TemplateProcessor.ProcessNode(OpenXmlCompositeElement rootElement)
   at DocxTemplater.DocxTemplate.Process()
```

**Comment by Amberg** [comment]
Do you still have problems?

if you use `template.BindModel("ds", htmlValue);` the template should be `{{ds}:html}` or `{{.}:html}`
`template.BindModel("ds", new { CLAUSES = htmlValue })` with `{{ds.CLAUSES}:html}` is correct too

Can you post the full stack trace?

**Comment by Amberg** [comment]
The  `DocxTemplate.Open(templatePath);` had a bug.

**Workaround for v2.3.1**
- Pass ProcessSettings.Default as second Argument ``DocxTemplate.Open(templatePath, ProcessSettings.Default);`
- Or use the constructor with a stream `new DocxTemplate(stream)`

**Fixed with v2.3.2**: https://github.com/Amberg/DocxTemplater/releases/tag/v2.3.2

**Comment by rst30** [comment]
Thank you, sorry for late response. Yes last time i'm trying this the error still persists. Will try the new version release and update again later.
